### ISSUE TITLE
step by step documentation

### ISSUE THREAD

**Issue by OURABIG** [AGulev/DefUnityAds#33]
i would like to add ads to my game but when i try this i get confused and i always have errors maybe if you can just make a step by step setup tutorial and how to show an ad like in a wiki or in a video and thanks for this amazing extension

**Comment by AGulev** [comment]
@OURABIG please share what exactly errors you have

**Comment by OURABIG** [comment]
> @OURABIG please share what exactly errors you have

well everything i can't even setup the ads or initialize them i just got confused to do it

**Comment by AGulev** [comment]
Did you see the example in this repo?

**Comment by OURABIG** [comment]
> Did you see the example in this repo?

yes of course and i didn't understand a lot it's a bit advance what should i create first

**Comment by AGulev** [comment]
you said :
>i always have errors 

What kind of errors do you have?
The install process is simple as:
- initialize
- wait when initialization is done
- show ads

**Comment by OURABIG** [comment]
> you said :
> 
> > i always have errors
> 
> What kind of errors do you have? The install process is simple as:
> 
>     * initialize
> 
>     * wait when initialization is done
> 
>     * show ads

so i add the direcrly in .gui_script no need to another script

**Comment by OURABIG** [comment]
> you said :
> 
> > i always have errors
> 
> What kind of errors do you have? The install process is simple as:
> 
>     * initialize
> 
>     * wait when initialization is done
> 
>     * show ads

i add initialize function and unity ads code then i go when button is clicked show ads is this what i need to do

**Comment by AGulev** [comment]
1. Initialize SDK with callback (third parameter `isDebug`):
https://github.com/AGulev/DefVideoAds/blob/abaeb8f6902457c638b3b08ab4e69ee567dd8b79/example/ui.gui_script#L217

Here is an example of the callback function:
https://github.com/AGulev/DefVideoAds/blob/abaeb8f6902457c638b3b08ab4e69ee567dd8b79/example/ui.gui_script#L76-L189

2. Wait for `unityads.EVENT_COMPLETED` in `unityads.MSG_INIT` :

https://github.com/AGulev/DefVideoAds/blob/abaeb8f6902457c638b3b08ab4e69ee567dd8b79/example/ui.gui_script#L79-L82

3. When it's done, you are able to use ADS functions. First of all you need to load ADS (in this function you use `placementId` of your AD as the first/only parameter):

https://github.com/AGulev/DefVideoAds/blob/abaeb8f6902457c638b3b08ab4e69ee567dd8b79/example/ui.gui_script#L341

4. When loading is done, you'll get `unityads.EVENT_LOADED` for message_id `unityads.MSG_LOAD` in the callback:

https://github.com/AGulev/DefVideoAds/blob/abaeb8f6902457c638b3b08ab4e69ee567dd8b79/example/ui.gui_script#L130-L134

5. Now you can show ADS (in this function you use `placementId` of your AD as the first/only parameter):

https://github.com/AGulev/DefVideoAds/blob/abaeb8f6902457c638b3b08ab4e69ee567dd8b79/example/ui.gui_script#L346

6. For the `show()` function you should use this `message_id` in the callback:
https://github.com/AGulev/DefVideoAds/blob/abaeb8f6902457c638b3b08ab4e69ee567dd8b79/example/ui.gui_script#L94-L109

**Comment by AGulev** [comment]
Also, you can take a look this [ads_wrapper](https://github.com/potatojam/defold-ads-wrapper) , maybe it will be easier for you.
